criu/page-xfer.c

```diff
@@ -846,10 +846,10 @@ int page_xfer_predump_pages(int pid, struct page_xfer *xfer,
 
 		bufvec.iov_base = userbuf;
 		bufvec.iov_len = bytes_read;
-		ret = vmsplice(ppb->p[1], &bufvec, 1, SPLICE_F_NONBLOCK);
+		ret = vmsplice(ppb->p[1], &bufvec, 1, 0);
```

Comment:
SPLICE_F_NONBLOCK is set intentionally. We know the size of this pipe and we call vmsplice exactly for this size. And nobody reads from this pipe when vmsplice is being called, so if this call will be blocked, it will be blocked forever.

Reply:
Thank you for explaining this!